lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -2419,5 +2412,41 @@ export class UserAgentApplication {
     return extractedUri;
   }
 
+  /**
+   * Utils function to create the Authentication
+   * @param userObject
+   * @param scopes
+   * @param silentCall
+   */
+  private getTokenType(userObject: User, scopes: string[], silentCall: boolean): string {
+
+    // if user is passed and matches the user object/or set to getUser() from cache
+    // if client-id is passed as scope, get id_token else token/id_token_token (in case no session exists)
+    let tokenType: string;
+
+    // acquireTokenSilent
+    if (silentCall) {
+      if (Utils.compareObjects(userObject, this.getUser())) {
+        tokenType = (scopes.indexOf(this.clientId) > -1) ? ResponseTypes.id_token : ResponseTypes.token;
+      }
+      else {
+        tokenType  = (scopes.indexOf(this.clientId) > -1) ? ResponseTypes.id_token : ResponseTypes.id_token_token;
+      }
+
+      return tokenType;
+    }
+    // all other cases
+    else {
+      tokenType = Utils.compareObjects(userObject, this.getUser()) ? ResponseTypes.id_token : ResponseTypes.id_token_token;
```

Comment:
This assignment is confusing. If the line below gets lost/updated, we will have issues.

Reply:
Made the needed changes. Please review and approve.